### PR TITLE
Sourcelink

### PR DESCRIPTION
**Pull request by nosami**

### CODE REVIEW

File: Mono.Debugging/Mono.Debugging.Client/SourceLocation.cs [hunk 1]
```diff
@@ -15,14 +17,15 @@ public class SourceLocation
...
 		public byte[] FileHash { get; private set; }
+		public SourceLink SourceLink { get; private set; }
 
 		[Obsolete]
 		public SourceLocation (string methodName, string fileName, int line)
 			: this (methodName, fileName, line, -1, -1, -1, null)
 		{
 		}
 
-		public SourceLocation (string methodName, string fileName, int line, int column, int endLine, int endColumn, byte[] hash = null)
```
Comment (nosami) [inline Mono.Debugging/Mono.Debugging.Client/SourceLocation.cs]: Fails binary compat test

```
  =================================
  These actual lines are new:
  In assembly 'MonoDevelop.Debugger.AD7, Version=2.1.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35': Unable to resolve member reference 'System.Void Mono.Debugging.Client.SourceLocation::.ctor(System.String,System.String,System.Int32,System.Int32,System.Int32,System.Int32,System.Byte[])' from assembly 'Mono.Debugging, Version=0.0.0.0, Culture=neutral, PublicKeyToken=5e9ce85b0923c84f'
  =================================
```

File: Mono.Debugging.Soft/PortablePdbData.cs [hunk 2]
```diff
@@ -63,6 +67,7 @@ public static bool IsPortablePdb (string pdbFileName)
 		public PortablePdbData (string pdbFileName)
 		{
 			this.pdbFileName = pdbFileName;
+			sourceLinkMaps = new Lazy<IEnumerable<SourceLinkMap>> (GetSourceLinkMaps);
```
Comment (Therzok) [inline Mono.Debugging.Soft/PortablePdbData.cs]: Can we use SourceLinkMap[] instead of IEnumerable?

Comment (Therzok) [reply Mono.Debugging.Soft/PortablePdbData.cs]: IEnumerable means lazy evaluation for me, and we have the data ready.

Comment (nosami) [reply Mono.Debugging.Soft/PortablePdbData.cs]: Changed.

**Pull request merged**